mantidproject/mantid
# Project Recovery: Remove saving of groups

**Description of work.**
Changed multiple classes to stop project recovery saving grouped workspaces, this is useful because it stops a situation where project recovery does not work properly, when you delete a workspace or move it from outside of a group

**To test:**
* Before opening Mantid, set `projectRecovery.secondsBetween = 10` in `Mantid.properties`
* Run the following script
```
CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', Names='A,B,C', Units='U,U,U', OutputWorkspace='ws')  
CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', Names='A,B,C', Units='U,U,U', OutputWorkspace='ws1')  
CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', Names='A,B,C', Units='U,U,U', OutputWorkspace='ws2')  
CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', Names='A,B,C', Units='U,U,U', OutputWorkspace='ws3') 
GroupWorkspaces(InputWorkspaces='ws,ws1,ws2,ws3', OutputWorkspace='NewGroup')
```
* Delete `ws3`
* Wait for the project to be saved
* Run `Segfault`
* Restart Mantid
* Click Yes on the project recovery pop up
* You should have all workspaces back outside of the original group

Fixes #22922 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
